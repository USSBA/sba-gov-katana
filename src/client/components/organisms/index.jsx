/* IMPORTS */

// miscellaenous organisms
import BusinessGuideTileCollection from "./business-guide-tile-collection/business-guide-tile-collection.jsx";
import CardCollection from "./card-collection/card-collection.jsx";
import DeveloperTester from "./developer-tester/developer-tester.jsx";
import DocumentCardCollection from "./document-card-collection/document-card-collection.jsx";
import FrontPageHero from "./front-page-hero/front-page-hero.jsx";
import FundingProgramsTileCollection from "./funding-programs-tile-collection/funding-programs-tile-collection.jsx";
import DisasterAlert from "./header-footer/disaster-alert/disaster-alert.jsx";
import NotificationBar from "./header-footer/notification-bar.jsx";
import Hero from "./hero/hero.jsx";
import RelatedDocumentCards from "./related-document-cards/related-document-cards.jsx";
import SearchBox from "./search-box/search-box.jsx";
import SectionNav from "./section-nav/section-nav.jsx";

// header-footer organisms
import Footer from "./header-footer/footer/footer.jsx";
import MobileNav from "./header-footer/mobile-nav/mobile-nav.jsx";
import Header from "./header-footer/header/header.jsx";

// homepage organisms
import Blog from "./homepage/blog/blog.jsx";
import FrontPageLady from "./homepage/front-page-lady/front-page-lady.jsx";
import HappeningNow from "./homepage/happening-now/happening-now.jsx";

// lender match organisms
// -- form
import AdditionalInfoForm from "./lender-match/form/additional-info.jsx";
import BusinessInfoForm from "./lender-match/form/business-info.jsx";
import ContactInfoForm from "./lender-match/form/contact-info.jsx";
import IndustryInfoForm from "./lender-match/form/industry-info.jsx";
import LoanInfo from "./lender-match/form/loan-info.jsx";
import ReviewSection from "./lender-match/form/review-page-helpers.jsx";
import ReviewSubmitInfoForm from "./lender-match/form/review-submit-info.jsx";
// -- landing page
import FindLendersIntro from "./lender-match/landing-page/find-lenders-intro.jsx";
import HelpfulQuestions from "./lender-match/landing-page/helpful-questions.jsx";
import {HowItWorksSection} from "./lender-match/landing-page/how-it-works.jsx";
import PreparationChecklist from "./lender-match/landing-page/preparation-checklist.jsx";
// -- success page
import ConfirmSection from "./lender-match/success-page/confirm-section.jsx";
import DynamicCounselingAndTools from "./lender-match/success-page/counseling-and-tools.jsx";
import CounselorMap from "./lender-match/success-page/counselor-map.jsx";
import EmailConfirmationInvalid from "./lender-match/success-page/email-confirmed-page.jsx";

// modal organisms
import LeaveSbaModal from "./modals/leave-sba-modal/leave-sba-modal.jsx";
import MobileSectionNavModal from "./modals/mobile-section-nav/mobile-section-nav.jsx";

/* EXPORTS */

// miscellaneous organisms
export {
	BusinessGuideTileCollection,
	CardCollection,
	DeveloperTester,
	DocumentCardCollection,
	FrontPageHero,
	FundingProgramsTileCollection,
	DisasterAlert,
	NotificationBar,
	Hero,
	RelatedDocumentCards,
	SearchBox,
	SectionNav
};

// header-footer organisms
export {
	Footer,
	MobileNav,
	Header
};

// homepage organisms
export {
	Blog,
	FrontPageLady,
	HappeningNow
};

// lender match organisms
export {
	AdditionalInfoForm,
	BusinessInfoForm,
	ContactInfoForm,
	IndustryInfoForm,
	LoanInfo,
	ReviewSection,
	ReviewSubmitInfoForm,
	FindLendersIntro,
	HelpfulQuestions,
	HowItWorksSection,
	PreparationChecklist,
	ConfirmSection,
	DynamicCounselingAndTools,
	CounselorMap,
	EmailConfirmationInvalid
};

// modal organisms
export {
	LeaveSbaModal,
	MobileSectionNavModal
};
