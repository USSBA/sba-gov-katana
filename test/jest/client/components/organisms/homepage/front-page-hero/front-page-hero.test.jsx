import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { FrontPageHero } from 'organisms'

const buttonProps = [
  {
    url: '/button-url',
    title: 'This is a button'
  }
]

const heroProps = {
  type: 'hero',
  caption: 'Want to be an entrepreneur?',
  imageLarge: {
    url: '/largeImage.jpg',
    alt: 'Start and grow your business',
    width: 1920,
    height: 768
  },
  imageMedium: {
    url: '/mediumImage.jpg',
    alt: 'medium image',
    width: 1446,
    height: 768
  },
  imageSmall: {
    url: '/smallImage.jpg',
    alt: 'small image',
    width: 1149,
    height: 1149
  },
  link: {},
  title: 'Start and grow your business'
}

describe('Front Page Hero', () => {
  test('should render correctly', () => {
    const tree = renderer.create(<FrontPageHero hero={heroProps} button={buttonProps[0]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // test('callout button renders correctly', () => {
  //   const component = shallow(<FrontPageHero hero={heroProps} button={buttonProps[0]} />)
  //
  //   expect(component.find('LargeInversePrimaryButton').props().id).toEqual('frontpage-hero-button')
  //   expect(component.find('LargeInversePrimaryButton').props().text).toEqual('This is a button')
  //   expect(component.find('LargeInversePrimaryButton').props().url).toEqual('/button-url')
  // })

  test('should load correct elements before rendering', () => {
    const component = shallow(<FrontPageHero hero={heroProps} button={buttonProps[0]} />)

    expect(component.find('heroLarge').root.unrendered.props.hero.title).toEqual(
      'Start and grow your business'
    )
    expect(component.find('heroLarge').root.unrendered.props.hero.caption).toEqual(
      'Want to be an entrepreneur?'
    )
    expect(component.find('heroLarge').root.unrendered.props.hero.imageLarge.url).toEqual('/largeImage.jpg')
    expect(component.find('heroLarge').root.unrendered.props.hero.imageLarge.alt).toEqual(
      'Start and grow your business'
    )
    expect(component.find('heroLarge').root.unrendered.props.hero.imageLarge.width).toEqual(1920)
    expect(component.find('heroLarge').root.unrendered.props.hero.imageLarge.height).toEqual(768)

    expect(component.find('heroMedium').root.unrendered.props.hero.imageMedium.url).toEqual(
      '/mediumImage.jpg'
    )
    expect(component.find('heroMedium').root.unrendered.props.hero.imageMedium.alt).toEqual('medium image')
    expect(component.find('heroMedium').root.unrendered.props.hero.imageMedium.width).toEqual(1446)
    expect(component.find('heroMedium').root.unrendered.props.hero.imageMedium.height).toEqual(768)

    expect(component.find('heroSmall').root.unrendered.props.hero.imageSmall.url).toEqual('/smallImage.jpg')
    expect(component.find('heroSmall').root.unrendered.props.hero.imageSmall.alt).toEqual('small image')
    expect(component.find('heroSmall').root.unrendered.props.hero.imageSmall.width).toEqual(1149)
    expect(component.find('heroSmall').root.unrendered.props.hero.imageSmall.height).toEqual(1149)
  })
})
