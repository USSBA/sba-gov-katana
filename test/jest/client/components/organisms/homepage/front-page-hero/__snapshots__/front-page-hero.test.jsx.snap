// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Front Page Hero should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="imageContainer"
  >
    <img
      alt="Start and grow your business"
      className="heroLarge"
      src="http://kevin.sba.fun/largeImage.jpg"
    />
    <img
      alt="medium image"
      className="heroMedium"
      src="http://kevin.sba.fun/mediumImage.jpg"
    />
    <img
      alt="small image"
      className="heroSmall"
      src="http://kevin.sba.fun/smallImage.jpg"
    />
  </div>
  <div
    className="boxContainer"
  >
    <div
      className="box"
    >
      <h1
        className="title"
      >
        Start and grow your business
      </h1>
      <div
        className="text"
      >
        Want to be an entrepreneur?
      </div>
      <a
        className="button primary spacing"
        onClick={[Function]}
        style={Object {}}
      >
        This is a button
      </a>
    </div>
  </div>
</div>
`;
